plasmapy/analysis/swept_langmuir/ion_saturation_current.py

```diff
@@ -0,0 +1,240 @@
+        +-----------------------+--------------------------------------+
+        | ``"exp_plus_linear"`` | 1.0                                  |
+        +-----------------------+--------------------------------------+
+
+        |
+
+        Cannot be used with keyword ``voltage_bound``.
+
+    voltage_bound: `float`
+        A bias voltage (in volts) that specifies an upper bound used to
+        collect the points for the curve fit.  That is, points that
+        satisfy ``voltage <= voltage_bound`` are used in the fit.
+        (DEFAULT ``None``)
+
+        |
+
+        Cannot be used with keyword ``current_bound``.
+
+    Returns
+    -------
+    isat: `~plasmapy.analysis.fit_functions.Linear`
+        A :term:`fit-function` representing the linear portion of the
+        fitted curve.
+
+    extras: `ISatExtras`
+        Additional information from the curve fit:
+
+        * ``extras.fitted_func`` is the :term:`fit-function` (specified
+          by ``fit_type``) fitted to the IV-curve
+        * ``extras.rsq`` is the coefficient of determination
+          (r-squared) value of the ``extras.fitted_func`` to the IV-curve
+        * ``extras.fitted_indices`` is a `slice` object representing the
+          points used in the curve fit (i.e.
+          ``(voltage[extras.fitted_indices], current[extras.fitted_indices])``).
+
+    Notes
+    -----
+    This routine works by:
+
+    1. Selecting the points to be used in the fit as determined by
+       ``voltage_bound`` or ``current_bound``.
+    2. Fitting the selected points with the :term:`fit-function`
+       specified by ``fit_type``.
+    3. Extracting the linear component of the fit and returning that as
+       the ion-saturation current.
+
+    This routine opts to return a function representing a linear
+    ion-saturation current, since, while ideal planar Langmuir probes
+    reach a steady-state ion-saturation current, real world Langmuir
+    probes "suffer" from expanding sheaths as the bias voltage
+    becomes increasingly negative.  This sheath expansion results the
+    ion-saturation current also increasing.
+    """
+    rtn_extras = ISatExtras(rsq=None, fitted_func=None, fitted_indices=None)._asdict()
+
+    _settings = {
+        "linear": {
+            "func": ffuncs.Linear,
+            "current_bound": 0.4,
+        },
+        "exp_plus_linear": {
+            "func": ffuncs.ExponentialPlusLinear,
+            "current_bound": 1.0,
+        },
+        "exp_plus_offset": {
+            "func": ffuncs.ExponentialPlusOffset,
+            "current_bound": 1.0,
+        },
+    }  # type: Dict[str, Dict[str, Any]]
+    try:
+        default_current_bound = _settings[fit_type]["current_bound"]
+        fit_func = _settings[fit_type]["func"]()
+        rtn_extras["fitted_func"] = fit_func
+    except KeyError:
+        raise ValueError(
+            f"Requested fit '{fit_type}' is not a valid option.  Valid options "
```

Comment:
```suggestion
            f"Requested fit {fit_type!r} is not a valid option.  Valid options "
```
[optional]

Reply:
Interesting little trick.  Not sure how I feel about it.  As a coder I like that it could help with consistency, but I also feel like that comes with a penalty of less explicitness.  I don't think this is the original intent behind `!r` and there might be other behaviors hidden behind the veil.

Reply:
Oh wow, yeah, that's the first time I'm seeing it as well! Definitely a nice trick, but the previous implementation is way more obvious.